hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/TestFsShellTouch.java

```diff
@@ -200,6 +200,72 @@ public void testTouch() throws Exception {
     FileStatus fileStatus = lfs.getFileStatus(newFile);
     assertThat(fileStatus.getAccessTime()).isEqualTo(dateObj.getTime());
     assertThat(fileStatus.getModificationTime()).isEqualTo(dateObj.getTime());
+
+    touchDirTests();
+  }
+
+  private void touchDirTests() throws Exception {
```

Comment:
How about creating a separate test method for directory?

Reply:
Yeah that wouldn't hurt either, will update the PR.